adapters/eea-direct.js

```diff
@@ -2,146 +2,104 @@ import { acceptableParameters, convertUnits } from '../lib/utils';
+  }
+}
 
-// formats data to match openAQ standard
-const formatData = (data, source, cb) => {
-  const stations = data[0];
+async function fetchMetadata (source) {
+  return request({url: stationsLink})
+    .pipe(JSONStream.parse('*'))
+    .pipe(new DataStream())
+    .filter(({stationId}) => stationId.startsWith(source.country))
+    .accumulate((acc, item) => (acc[item.stationId] = item), {});
+}
 
-  // the CSV files contain 48 hours worth of data
-  // filter this down to records that were inserted up to two hours ago
-  // this vastly reduces the amount of redundant inserts fetch tries to make
-  const timeLastInsert = data[1].reduce((a, b) => Math.max(a, Date.parse(b.value_datetime_inserted) || 0), 0);
-  const records = data[1].filter(o => Date.parse(o.value_datetime_inserted) > (timeLastInsert - 3700000));
+function fetchPollutants (source, stations) {
+  const pollutants = acceptableParameters.map(
+    (pollutant) => pollutant === 'pm25' ? 'PM2.5' : pollutant.toUpperCase()
+  );
 
-  map(records, (record, done) => {
-    const matchedStation = stations.find(station => station.stationId === record['station_code']);
-    if (!(matchedStation)) {
-      return done(null, {});
-    }
-    const timeZone = tzlookup(matchedStation.latitude, matchedStation.longitude);
-    let m = {
-      location: record['station_code'],
-      city: matchedStation.city ? matchedStation.city : (
-        matchedStation.location ? matchedStation.location : source.city
-      ),
-      coordinates: {
-        latitude: Number(matchedStation.latitude),
-        longitude: Number(matchedStation.longitude)
-      },
-      parameter: record['pollutant'].toLowerCase().replace('.', ''),
-      date: makeDate(record['value_datetime_end'], timeZone),
-      value: Number(record['value_numeric']),
-      unit: record['value_unit'],
-      attribution: [{
-        name: 'EEA',
-        url: source.sourceURL
-      }],
-      averagingPeriod: {
-        unit: 'hours',
-        value: 1
-      }
-    };
-    // apply unit conversion to generated record
-    done(null, convertUnits([m])[0]);
-  }, (err, measurements) => {
-    if (err) {
-      return cb(null, {name: 'unused', measurements: []});
-    }
-    cb(null, {name: 'unused', measurements: measurements});
-  });
-};
+  return new MultiStream(
+    pollutants.map(pollutant => {
+      const url = source.url + source.country + '_' + pollutant + '.csv';
+      const timeLastInsert = Date.now() - 72e5;
+
+      return new StringStream()
+        .use(stream => {
+          const resp = request.get({url})
+            .on('response', ({statusCode}) => {
+              +statusCode !== 200
+                ? stream.end()
+                : resp.pipe(stream);
+            });
+          return stream;
+        })
+        .CSVParse({header: true, delimiter: ',', skipEmptyLines: true})
+        // TODO: it would be good to provide the actual last fetch time so that we can filter already inserted items in a better way
```

Comment:
Good point! I believe we're actually storing this in Redis for another purpose so we may be able to get this here without too much difficulty, but I think this is an ok thing to leave undone for now.